### PR TITLE
fix(sdk): resolve some React 19+ issues in our testing tools

### PR DESCRIPTION
**Pull request by byCedric**
# Why

There are a couple:

**1. React 19 deprecates `react-test-renderer`**

Just the deprecation warning alone causes a few tests to fail. Unfortunately, there is no replacement yet for React Native, meaning even if we did switch to `@testing-library/react-native`, we'd still use `react-test-renderer`.

> Luckily for us, there is a temporary workaround, which I've added to `jest-expo` for now. Read more:  https://discord.com/channels/514829729862516747/514832743654228009/1266527040480477274

**2. `@testing-library/react-hooks` seems to be silently deprecated.**

There are no updates for React 18, or 19. For React 18, they mention to swap to `@testing-library/react` itself, and use the `renderHook` function the original library. That's why I pulled out `@testing-library/react-hooks` from our dependencies.

**3. Due to RTR being deprecated, we need to move over to `@testing-library/react-native`.**

Unfortunately, we'd have to wait for proper React 19 support in https://github.com/callstack/react-native-testing-library. Right now, it seems that `react-test-renderer` returns `null` for a lot of renders, which also seems to indicate there are issues in the current canary version we are using.

# How

Unfortunately, this won't fix the current test failures fully. But it does get us on a better path towards being green again (with `react-test-renderer@19`)

# Test Plan

See CI

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/jest-expo/src/preset/setup-react-19.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+// This is a temporary workaround for `react-test-renderer@19+`.
+// The renderer is deprecated, but there is no replacement for React Native yet.
+// Setting this global property disables the deprecation warning, and changes RTR to legacy mode.
+// See: https://discord.com/channels/514829729862516747/514832743654228009/1266527040480477274
+globalThis.IS_REACT_NATIVE_TEST_ENVIRONMENT = true;
```
Comment (byCedric) [inline packages/jest-expo/src/preset/setup-react-19.js:5]: We could also define this in the `expo-module-scripts` preset, if we don't want to ship this temporary workaround to users.

**Review by Kudo — APPROVED** [review]
looks good to me for expo-sqlite changes.

**Review by Simek — APPROVED** [review]
Looks good from the docs app side, thanks! 👍

**Review by tsapeta — APPROVED** [review]

File: packages/jest-expo/src/preset/setup-react-19.js (hunk 1 above)
Comment (byCedric) [reply packages/jest-expo/src/preset/setup-react-19.js:5]: Edit: I'll pull it out of `jest-expo` and into `expo-module-scripts`, we need a user-facing solution eventually and this is just a temporary workaround.

**Pull request merged**